### ISSUE TITLE
Claims owned by players who've not logged in for a while can't be deleted with /deleteallclaims

### ISSUE THREAD

**Issue by Cassolotl** [GriefPrevention/GriefPrevention#238]
### What behaviour is observed:
I am the admin. A player who has not logged in for a long time still had griefprevention claims. I tried to delete all of their claims with `/deleteallclaims username`. I got an error message, that said "No player by that name has logged in recently." If I stood inside that claim and typed `/abandonclaim` the claim was deleted without any problems.

Edit: I would also like to be able to do `/AdjustBonusClaimBlocks <username> 0` for players who've not logged in for a while, but that also gets me a "No player by that name has logged in recently."
### What behaviour is expected:
All of that player's claims being deleted.
### Steps/models to reproduce:
Type `/deleteallclaims (username)` for someone who's not logged in for a while, but whose claims are still active.

### Spigot/Craftbukkit/Paper and GriefPrevention version:
[23:40:40] [Server thread/INFO]: This server is running CraftBukkit version git-Spigot-65e8124-963b19c (MC: 1.12.1) (Implementing API version 1.12.1-R0.1-SNAPSHOT)
[23:40:40] [Server thread/INFO]: Checking version, please wait...
[23:40:41] [Thread-218/INFO]: You are 73 version(s) behind
[23:41:36] [Server thread/INFO]: GriefPrevention version 16.7.1

### Stack trace/error

```
Paste entire stack trace inside here (don't remove the backticks!)
If not applicable, paste server.log at gist.github.com and paste the link here
```

I'm not sure what that is, but this is what shows in the console when I type `/deleteallclaims (username)`:

```
[23:42:09] [Server thread/INFO]: [GriefPrevention] §cNo player by that name has logged in recently.
```

### GriefPrevention config.yml (if applicable)

```yml

# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/
GriefPrevention:
  SeaLevelOverrides:
    world: -1
    world_nether: -1
    world_the_end: -1
    world_creative: -1
  Claims:
    Mode:
      world: Survival
      world_nether: Survival
      world_creative: Survival
      world_the_end: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: false
    EnderPearlsRequireAccessTrust: true
    ProtectHorses: true
    InitialBlocks: 100000
    Claim Blocks Accrued Per Hour:
      Default: 1000
    Max Accrued Claim Blocks:
      Default: 800000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 0
    ExtendIntoGroundDistance: 5
    MinimumWidth: 2
    MinimumArea: 4
    MaximumDepth: 0
    InvestigationTool: STICK
    ModificationTool: WOOD_SPADE
    Expiration:
      ChestClaimDays: 365
      UnusedClaimDays: 365
      AllClaims:
        DaysInactive: 365
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    PortalGenerationRequiresPermission: false
    VillagerTradingRequiresPermission: false
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
  Spam:
    Enabled: false
    LoginCooldownSeconds: 0
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/tell;/global;/local;/w;/msg;/r;/t
    WhisperSlashCommands: /tell;/pm;/r;/w;/t;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: true
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 60
    Logout Message Delay In Seconds: 10
  PvP:
    RulesEnabledInWorld:
      world: true
      world_nether: true
      world_the_end: true
      world_creative: true
    ProtectFreshSpawns: true
    PunishLogout: true
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: false
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: false
      AdministrativeClaims: false
      AdministrativeSubdivisions: false
    AllowLavaDumpingNearOtherPlayers: true
    AllowFlintAndSteelNearOtherPlayers: true
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksPurchaseCost: 0.0
    ClaimBlocksSellValue: 0.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: false
  BlockSurfaceOtherExplosions: true
  LimitSkyTrees: true
  LimitTreeGrowth: false
  LimitPistonsToLandClaims: false
  FireSpreads: true
  FireDestroys: true
  AdminsGetWhispers: false
  AdminsGetSignNotifications: true
  SmartBan: true
  Mute New Players Using Banned Words: true
  MaxPlayersPerIpAddress: 3
  Siege:
    Worlds: []
    BreakableBlocks:
    - DIRT
    - GRASS
    - LONG_GRASS
    - COBBLESTONE
    - GRAVEL
    - SAND
    - GLASS
    - THIN_GLASS
    - WOOD
    - WOOL
    - SNOW
  EndermenMoveBlocks: true
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: false
  HardModeZombiesBreakDoors: false
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Mods:
    BlockIdsRequiringAccessTrust: []
    BlockIdsRequiringContainerTrust:
    - 227:*:Battery Box
    - 130:*:Transmutation Tablet
    - 128:*:Alchemical Chest and Energy Condenser
    - 181:*:Various Chests
    - 178:*:Ender Chest
    - 150:*:Various BuildCraft Gadgets
    - 155:*:Filler
    - 157:*:Builder
    - 158:*:Template Drawing Table
    - 126:*:Various EE Gadgets
    - 138:*:Various RedPower Gadgets
    - 137:*:BuildCraft Project Table and Furnaces
    - 250:*:Various IC2 Machines
    - 161:*:BuildCraft Engines
    - 169:*:Automatic Crafting Table
    - 177:*:Wireless Components
    - 183:*:Solar Arrays
    - 187:*:Charging Benches
    - 188:*:More IC2 Machines
    - 190:*:Generators, Fabricators, Strainers
    - 207:*:Computer
    - 208:*:Computer Peripherals
    - 246:*:IC2 Generators
    - 24303:*:Teleport Pipe
    - 24304:*:Waterproof Teleport Pipe
    - 24305:*:Power Teleport Pipe
    - 4311:*:Diamond Sorting Pipe
    - 216:*:Turtle
    BlockIdsExplodable:
    - 161:*:BuildCraft Engines
    - 246:5:Nuclear Reactor
    PlayersIgnoringAllClaims: []
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: true
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false

```

### Plugin list (if applicable):

[23:44:19] [Server thread/INFO]: Plugins (9): CoreProtect, PermissionsEx, dynmap, MultiWorld, AutismChat3, Seen, GriefPrevention, PvPManager, Floristics

**Comment by RoboMWM** [comment]
Yea, that's because GP doesn't load playerdata for players that haven't logged in for over 90 days. Not sure of an easy way to fix this other than just loading every player that has ever played on the server/is in the usercache. I could also try seeing if UUID input could work as well (then it'd just do a direct check for the file in the data folder).

**Comment by Cassolotl** [comment]
It's kind of a pain because it means claims don't automatically expire, I think! We have a player who's not logged in for over 2 years whose claims still have to be admin-deleted when someone finds one, and that's just started happening with a more recently banned player too.

**Comment by RoboMWM** [comment]
> because it means claims don't automatically expire, I think!

Nope - the claim expiry check iterates through all claims (slowly over time). Which makes sense - no need to iterate through every single player that has ever logged in (as there's no guarantee that they own any claims).

**Comment by Cassolotl** [comment]
Ohhh, huh, okay - then there is a separate bug preventing claims expiring. But I don't know how to reproduce that so we can ignore it for now. :)

**Comment by Jikoo** [comment]
They likely have over 10k blocks available or 5k bonus blocks. From your config:
```YAML
GriefPrevention:
  Claims:
    Expiration:
      AllClaims:
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
```

**Comment by Cassolotl** [comment]
@Jikoo 

>They likely have over 10k blocks available or 5k bonus blocks.

That prevents claims from expiring? How do I fix that?

**Comment by Jikoo** [comment]
> That prevents claims from expiring? How do I fix that?

You can fix it by reading the entire post :p I included the relevant settings.

**Comment by Cassolotl** [comment]
I read the entire post. You didn't say what I should change the settings to to fix the problem! Should I just remove those lines altogether? Or is there a command I can use to revoke someone's blocks and bonus blocks, if I only want to make that one person's claims expire?

Should I open an issue to ask for a setting that automatically deletes all of a player's available and bonus claim blocks when they're banned?

Edit: I tried `/AdjustBonusClaimBlocks <username> 0`, and I got:

`[17:56:01] [Server thread/INFO]: [GriefPrevention] §cNo player by that name has logged in recently.`

So this all loops around to the same issue - if a player hasn't logged in recently you can't delete all their claims at once, and you can't delete their available and bonus blocks to make sure their claims expire automatically, so each claim must be deleted manually by an admin. If they're banned they can't log in again to bodge it and make the system recognise them again.

**Comment by Jikoo** [comment]
This is regarding your longterm player whose claims you want removed, not the banned one. If the settings aren't to your liking, change them so they work for you. If you want claims of players with 10k blocks to be deleted, raise the cap. There's no infinite option unfortunately.

If you can't figure out what setting will satisfy you, how can I? You know your server and your needs far better than I ever could :p

Say you want players' claims to expire no matter what, just set those settings out of reach. 1000000 should be pretty prohibitive, and if you really want you could go even higher.

**Comment by Cassolotl** [comment]
This isn't about me not knowing what I want, this is about not knowing from the information you gave me how to make it happen.

So like, now I know that there's no "delete claims no matter what" setting (like changing the number to 0 or -1) and no way to delete a player's available and bonus blocks if they've not logged in for a while, I have some ideas for some new issues...

**Comment by RoboMWM** [comment]
>you can't delete their available and bonus blocks to make sure their claims expire automatically,

Only if they have exceeded the total or bonus claim blocks as listed in your config. For players not in memory, the claim expiry check temporarily loads the owner of the randomly-selected claim if it is not already in memory temporarily, which is why it is able to expire claims of players who have been inactive.

**Comment by Cassolotl** [comment]
>For players not in memory, the claim expiry check temporarily loads the owner of the randomly-selected claim if it is not already in memory temporarily

Would it be possible to change the plugin so that it does this when you want to delete all claims by a player or delete remaining available/bonus claim blocks, too?

**Comment by RoboMWM** [comment]
> Would it be possible to change the plugin so that it does this when you want to delete all claims by a player or delete remaining available/bonus claim blocks, too?

I guess that makes sense, considering the options we have to deem if we should skip a player or not is based on total/bonus blocks obtained.

**Comment by Cassolotl** [comment]
`/claimslist playername` also results in "No player by that name has logged in recently." This is really inconvenient; I can't even find their claims to delete manually one by one.

**Comment by seansbar** [comment]
Also /untrust a player that has not logged in in a long time also produces: "No player by that name has logged in recently."

Is there a way to load the old player in memory temporary to preform the needed commands?

**Comment by ScuroK** [comment]
i have the same problem. @RoboMWM  can you please fix this??

**Comment by RoboMWM** [comment]
@Cassolotl @ScuroK @seansbar You can now adjust the max amount of time a player can be inactive to be included in the offlineplayer cache as per the referenced commit, available in 16.11.2.

**Comment by Cassolotl** [comment]
@RoboMWM Thank you very much! :)
